### ISSUE TITLE
Exponentiated gradient descent

### ISSUE THREAD

**Issue by fjanoos** [pytorch/pytorch#30386]
## 🚀 Feature
I am interested in contributing an [exponentiated gradient descent optimizer](https://ttic.uchicago.edu/~tewari/lectures/lecture4.pdf)  to pytorch.

## Motivation

Optimizing variables / parameters with a simplex constraint on the parameters.

## Pitch

Contribute a new optimizer

## Alternatives

Projected gradient descent.

**Comment by gchanan** [comment]
@vincentqb can you make a call and remove the "needs research" label?

**Comment by vincentqb** [comment]
Though EG has been around since [1997](https://users.soe.ucsc.edu/~manfred/pubs/J36.pdf), I do not find strong evidence of EG being used in deep learning.

@fjanoos @bamos -- Do you have other references?

Tagging #22281 for constrained optimization

**Comment by nick11roberts** [comment]
Hello! I just came across this issue and thought I'd contribute a reference. [1] is a state-of-the-art Neural Architecture Search method that uses exponentiated gradient descent for relaxed architecture parameters. Selfishly, this would also be a very useful optimizer for my purposes. Thanks, and I hope this makes it into PyTorch someday! 

[1] https://arxiv.org/pdf/2004.07802.pdf
